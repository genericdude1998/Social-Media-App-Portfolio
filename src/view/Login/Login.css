* {
    font-family:Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif !important;
}

.baseLogin{
    height: 100vh;
    width: 100vw;
    min-width: 600px;
    background-color: lightblue;
}