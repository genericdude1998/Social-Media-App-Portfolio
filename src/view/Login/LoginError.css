*{
    box-sizing: border-box;
    padding: 0px; 
}

.baseError{
    text-align: center;
    color: red;
    font-weight: bold;
    font-size: 10px;
    margin: 10px;
}