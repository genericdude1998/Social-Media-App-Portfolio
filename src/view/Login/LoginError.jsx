import React from 'react';
import PropTypes from 'prop-types';
import styles from '../Login/LoginError.css'

const LoginError = ({error}) => {
    return (
        <div className={styles.baseError}>
            {error ? error.errorMessage: null}
        </div>
    );
}

LoginError.propTypes = {
    error: PropTypes.object,
}

export default LoginError;
