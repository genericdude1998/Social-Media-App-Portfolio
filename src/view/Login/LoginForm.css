*{
  box-sizing: border-box;
  padding: 0px; 
}

.baseForm{
    height: 50vh;
    width: 100vw;
    min-width: 600px;
    align-items:center;
    display: flex;
    flex-direction: column;
    justify-content: center;
}
.baseLabel{
    color: black;
    margin: 10px;
    text-align: center;
    font-weight: bold;
}
.baseInput{
   margin:5px;
   border: 0px;
   border: 1px solid black;

}
.baseButton{
    margin:20px;
    height: 20px;
    width: 50px;
    border: 1px solid black;
    background-color: white;
}