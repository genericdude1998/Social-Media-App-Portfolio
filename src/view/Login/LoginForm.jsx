import React from 'react';
import styles from './LoginForm.css';
import PropTypes from 'prop-types';

const LoginForm = ({
    username,
    password,
    onChangeUsername,
    onChangePassword,
    onSubmitLogin,
}) => {
    return (
        <form className={styles.baseForm} onSubmit={onSubmitLogin(username, password)}>
            <label className={styles.baseLabel}>Username:</label>
            <input type='text' className={styles.baseInput} onChange={onChangeUsername}/>
            <label className={styles.baseLabel}>Password:</label>
            <input type='password'className={styles.baseInput} onChange={onChangePassword}/>
            <button type='submit' className={styles.baseButton}>Login</button>
        </form>
    );
}

LoginForm.propTypes = {
    onChangeUsername: PropTypes.func.isRequired,
    onChangePassword: PropTypes.func.isRequired,
}

export default LoginForm;
