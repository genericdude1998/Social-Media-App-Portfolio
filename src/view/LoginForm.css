*{
  box-sizing: border-box;
  padding: 0px; 
}

.baseForm{
    background-image: url('/assets/background-login.jpg');
    background-size: cover;
    height: 100vh;
    width: 100vw;
    align-items:center;
    display: flex;
    flex-direction: column;
    height: 100vh;
    justify-content: center;
}
.baseLabel{
    color: black;
    margin: 10px;
    text-align: center;
    font-weight: bold;
}
.baseInput{
   margin:5px;
   border: 0px;
   border: 1px solid black;

}
.baseButton{
    margin:20px;
    height: 20px;
    width: 50px;
    border: 1px solid black;
    background-color: white;
}

.light{

}

.dark{

}