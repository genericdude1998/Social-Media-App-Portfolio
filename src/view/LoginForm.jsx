import React from 'react';
import styles from './LoginForm.css'

const LoginForm = () => {
    const [username, setUsername] = React.useState('');
    const [password, setPassword] = React.useState('');

    return (
        <form className={styles.baseForm}>
            <label className={styles.baseLabel}>Username:</label>
            <input type='text' className={styles.baseInput}/>
            <label className={styles.baseLabel}>Password:</label>
            <input type='password'className={styles.baseInput}/>
            <button type='submit' className={styles.baseButton}>Login</button>
        </form>
    );
}

export default LoginForm;
