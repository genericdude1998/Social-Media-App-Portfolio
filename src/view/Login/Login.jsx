import React from 'react';
import PropTypes from 'prop-types';
import styles from './Login.css';
import LoginForm from './LoginForm';
import LoginHeader from './LoginHeader';
import LoginFooter from './LoginFooter';
import { connect } from 'react-redux';
import { mapLoginDispatchToProps, mapLoginStateToProps } from '../mappers/mappers';
import LoginError from './LoginError';

export const Login = ({
    username,
    password,
    token,
    error,
    onChangePassword,
    onChangeUsername,
    onSubmitLogin,
}) => {
    return (
        <div className={styles.baseLogin}>
            <LoginHeader />
            <LoginForm 
                username={username} 
                password={password} 
                onChangeUsername={onChangeUsername} 
                onChangePassword={onChangePassword}
                onSubmitLogin={onSubmitLogin}
            />
            <LoginError error={error}/>
            <LoginFooter />
        </div>
    );
}

Login.propTypes = {
    username: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired,
    token: PropTypes.string,
    onChangeUsername: PropTypes.func.isRequired,
    onChangePassword: PropTypes.func.isRequired,
    onSubmitLogin: PropTypes.func.isRequired,
}

const ConnectedLogin = connect(mapLoginStateToProps, mapLoginDispatchToProps)(Login);

export default ConnectedLogin;
