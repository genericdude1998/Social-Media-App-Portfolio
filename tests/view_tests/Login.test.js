import React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../../src/view/Login/Login';
import LoginForm from '../../src/view/Login/LoginForm';

const mockOnChangePassword = jest.fn(); 
const mockOnChangeUsername = jest.fn(); 


let loginWrapper;
beforeEach(() => {
    loginWrapper = shallow(
        <Login
            username=''
            password=''
            token={null}
            onChangePassword= {mockOnChangePassword}
            onChangeUsername= {mockOnChangeUsername}
        />
    );
});

describe('Login', () => {
    it('should pass onChange handlers to LoginForm component', () => {
        const {onChangeUsername, onChangePassword} = loginWrapper.find(LoginForm).props();
        expect(onChangeUsername).toBe(mockOnChangeUsername);
        expect(onChangePassword).toBe(mockOnChangePassword);
    });
    it('should matchsnapshot', () => {
        expect(loginWrapper).toMatchSnapshot();
    });
});
