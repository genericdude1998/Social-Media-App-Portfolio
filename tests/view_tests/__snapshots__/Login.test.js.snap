// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should matchsnapshot 1`] = `
<div
  className="baseLogin"
>
  <LoginHeader />
  <LoginForm
    onChangePassword={[MockFunction]}
    onChangeUsername={[MockFunction]}
    onSubmitLogin={[MockFunction]}
    password=""
    username=""
  />
  <LoginError />
  <LoginFooter />
</div>
`;
